.github/workflows/ci.yml

```diff
@@ -28,43 +28,89 @@ jobs:
       - run: python -m pip install .[test]
       - run: ${{ matrix.lint-command }}
 
-  tests:
-    name: Python ${{ matrix.python-version }}, Django ${{ matrix.django-version }}, with ${{ matrix.database }}
+  sqlite-tests:
+    name: SQLite - Python ${{ matrix.python-version }}, Django ${{ matrix.django-version }}
     runs-on: ubuntu-22.04
     strategy:
+      fail-fast: false
       matrix:
         python-version: ["3.8", "3.9", "3.10", "3.11", "3.12", "3.13"]
         django-version: ["4.2", "5.0", "5.1", "5.2"]
-        database: ["postgresql", "sqlite", "postgresql-psycopg3", "postgresql-contenttypes"]
         exclude:
-          # Python 3.8 only supports Django 4.2
+          # Python 3.8/3.9 only supports Django 4.2
           - python-version: "3.8"
             django-version: "5.0"
           - python-version: "3.8"
             django-version: "5.1"
           - python-version: "3.8"
             django-version: "5.2"
-          # Python 3.9 only supports Django 4.2
           - python-version: "3.9"
             django-version: "5.0"
           - python-version: "3.9"
             django-version: "5.1"
           - python-version: "3.9"
             django-version: "5.2"
-          # Psycopg3 only for Python 3.11+
+
+    steps:
+      - uses: actions/checkout@v5
+      - name: Set TOX_ENV
+        run: echo "TOX_ENV=py$(echo ${{ matrix.python-version }} | tr -d .)-django$(echo ${{ matrix.django-version }} | tr -d .)-sqlite" >> $GITHUB_ENV
+
+      - name: Set up Python ${{ matrix.python-version }}
+        uses: actions/setup-python@v6
+        with:
+          python-version: ${{ matrix.python-version }}
+          allow-prereleases: true
+          cache: pip
+
+      - name: Install dependencies
+        run: |
+          python -m pip install --upgrade pip setuptools wheel tox
+
+      - name: Run tox targets for ${{ matrix.python-version }}
+        run: tox run -e ${{ env.TOX_ENV }}
+
+      - name: Upload coverage data
+        uses: actions/upload-artifact@v4
+        with:
+          name: coverage-data-${{ env.TOX_ENV }}
+          include-hidden-files: true
+          path: '.coverage.*'
+          if-no-files-found: ignore
+
+  postgresql-tests:
+    name: PostgreSQL - Python ${{ matrix.python-version }}, Django ${{ matrix.django-version }}, ${{ matrix.database }}
+    runs-on: ubuntu-22.04
+    strategy:
+      fail-fast: false
+      matrix:
+        python-version: ["3.8", "3.9", "3.10", "3.11", "3.12", "3.13"]
+        django-version: ["4.2", "5.0", "5.1", "5.2"]
+        database: ["postgresql", "postgresql-psycopg3"]
```

Comment:
nit: it's not about `database` anymore, but i do not have a better name now (maybe `use-case`?)